tempesta_fw/lb/tfw_lb_dummy.c

```diff
@@ -0,0 +1,204 @@
+/**
+ *		Tempesta FW
+ *
+ * The dummy load balancer suitable for testing and debugging.
+ *
+ * It supports only one connection to a single backend server.
+ * It sends all messages to this single connection without actually balancing
+ * anything. The connection is established only once upon start, and not
+ * restored automatically in case it is closed (you have to restart Tempesta FW
+ * to re-connect to the backend server).
+ *
+ * Also the module provides primitive configuration: only one "backend" entry is
+ * allowed in the configuration file (e.g. "backend 127.0.0.1:8080"). Multiple
+ * backends are not allowed.
+ *
+ * Copyright (C) 2015 Tempesta Technologies.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#include <net/tcp_states.h>
+
+#include "addr.h"
+#include "lb_mod.h"
+#include "log.h"
+#include "connection.h"
+#include "tempesta.h"
+
+MODULE_AUTHOR(TFW_AUTHOR);
+MODULE_DESCRIPTION("Tempesta FW dummy load balancer");
+MODULE_VERSION("0.0.1");
+MODULE_LICENSE("GPL");
+
+static struct socket *tfw_lb_dummy_be_sock;
+static TfwAddr tfw_lb_dummy_be_addr;
+
+static int
+tfw_lb_dummy_send_msg(TfwMsg *msg)
+{
+	struct sock *sk;
+
+	BUG_ON(!tfw_lb_dummy_be_sock);
+	sk = tfw_lb_dummy_be_sock->sk;
+
+	if (sk->sk_state != TCP_ESTABLISHED) {
+		TFW_ERR_ADDR("not connected to backend", &tfw_lb_dummy_be_addr);
```

Comment:
Is this really an error and we can't handle it? I guess we should just print a warning here and try to reestablish the connection....